backend/internal/server/apis/league/league.go

```diff
@@ -25,3 +27,23 @@ func getLeaguesHandler(c *fiber.Ctx) error {
 
 	return responder.OkWithData(c, leagues)
 }
+
+func postLeaguesHandler(c *fiber.Ctx) error {
+	log := locals.Logger(c)
+
+	leaguesRequest := &models.League{}
+	err := c.BodyParser(leaguesRequest)
+	if err != nil {
+		log.WithErr(err).Alert("Failed to parse leagues request payload")
+		return responder.BadRequest(c, "Failed to parse leagues request payload")
+	}
+
+	db := db.GetSession(c)
+	err = db.CreateLeagues(leaguesRequest)
+	if err != nil {
+		log.WithErr(err).Alert("Failed to save leagues request")
+		return responder.InternalServerError(c)
+	}
+
+	return responder.Ok(c)
```

Comment:
This needs to be `responder.OkWithData(c)`. I believe that even though this is a POST we still need to respond with OkWithData. See line 28.

Reply:
Nevermind, I see that all POST functions return Ok(c).